cache/contenthash/checksum.go

```diff
@@ -692,34 +692,35 @@ func shouldIncludePath(
 	parentDir *includedPath,
 ) (bool, error) {
 	var (
-		m   bool
-		err error
+		m         bool
+		matchInfo fileutils.MatchInfo
+		err       error
 	)
 	if includePatternMatcher != nil {
 		if parentDir != nil {
-			m, err = includePatternMatcher.MatchesUsingParentResult(candidate, parentDir.matchedIncludePattern)
+			m, matchInfo, err = includePatternMatcher.MatchesUsingParentResults(candidate, parentDir.includeMatchInfo)
 		} else {
-			m, err = includePatternMatcher.MatchesOrParentMatches(candidate)
+			m, matchInfo, err = includePatternMatcher.MatchesUsingParentResults(candidate, fileutils.MatchInfo{})
 		}
 		if err != nil {
 			return false, errors.Wrap(err, "failed to match includepatterns")
 		}
-		maybeIncludedPath.matchedIncludePattern = m
+		maybeIncludedPath.includeMatchInfo = matchInfo
 		if !m {
 			return false, nil
 		}
 	}
 
 	if excludePatternMatcher != nil {
 		if parentDir != nil {
-			m, err = excludePatternMatcher.MatchesUsingParentResult(candidate, parentDir.matchedExcludePattern)
+			m, matchInfo, err = includePatternMatcher.MatchesUsingParentResults(candidate, parentDir.excludeMatchInfo)
```

Comment:
This should be `excludePatternMatcher`, not `includePatternMatcher`. Same two lines below. Didn't catch this in the first review.

Reply:
Unfortunately, there is a bug in `MatchesUsingParentResults`. I opened https://github.com/moby/moby/pull/43047 to fix it. We'll need to get that merged and then revendor.